src/Internal/Test/ContractTest.purs

```diff
@@ -60,6 +74,30 @@ newtype ContractTestPlan = ContractTestPlan
     -> r
   )
 
+instance Semigroup ContractTestPlan where
+  append
+    (ContractTestPlan runContractTestPlan)
+    (ContractTestPlan runContractTestPlan') =
+    do
+      runContractTestPlan \distr tests -> do
+        runContractTestPlan'
+          \distr' tests' -> ContractTestPlan \h -> h (distr /\ distr') do
+            mapTest (_ <<< fst) tests
+            mapTest (_ <<< snd) tests'
+
+-- | Group `ContractTestPlans` together, so that they can be ran in the same Plutip instance
+groupContractTestPlans
```

Comment:
I question the usefulness of this function. It only allows two `ContractTestPlan`s to be joined, but requires a title. In real life scenarios, will users be forced to provide 2 titles to join 3 tests?

Reply:
Maybe we want an array of tests, grouped under a common title?
But then, couldn't the user just use

```
 runPlutipTestPlan config do
  sameWallets distribution $
     group "foo" do
        test "bar" do ...
        test "baz" do ...
```

?

I see no advantage in using an array, so maybe we should just remove this function?

Reply:
That method works, and that is what I actually have been using. This is was an idea I had that I carried over, but I will remove this since I don't actually use it myself.